crates/pera-core/src/authority.rs

```diff
@@ -1548,47 +1546,51 @@ impl AuthorityState {
         if !self.is_validator(epoch_store) {
             return Ok(());
         }
-        let status = match &effects {
-            TransactionEffects::V1(effects) => effects.status(),
-            TransactionEffects::V2(effects) => effects.status(),
-        };
+
+        let status = effects.status();
         if !status.is_ok() {
             // TODO (#303): Decide what to do about a failed transaction with MPC events.
             // If the transaction failed, we don't need to handle MPC events.
             return Ok(());
         }
-        let party_id = authority_name_to_party_id(&epoch_store.name, &epoch_store)?;
-        let mut dwallet_mpc_outputs_manager =
+
+        let party_id = authority_name_to_party_id(&epoch_store.name, epoch_store)?;
+        let mut dwallet_mpc_outputs_verifier =
             epoch_store.get_dwallet_mpc_outputs_verifier().await?;
+
         for event in &inner_temporary_store.events.data {
             if LockedNextEpochCommitteeEvent::type_() == event.type_ {
-                info!("received LockedNextEpochCommitteeEvent successfully");
-                dwallet_mpc_outputs_manager.completed_locking_next_committee = true;
+                info!("Received LockedNextEpochCommitteeEvent successfully");
+                dwallet_mpc_outputs_verifier.completed_locking_next_committee = true;
                 continue;
             }
-            let Ok(Some(session_info)) = session_info_from_event(
+
+            // Attempt to extract session info; skip invalid events.
+            let session_info = match session_info_from_event(
                 event,
                 party_id,
-                dwallet_mpc_outputs_manager.network_key_version(),
-            ) else {
-                continue;
+                dwallet_mpc_outputs_verifier.network_key_version(),
+            ) {
+                Ok(Some(info)) => info,
+                _ => continue,
             };
+
+            // todo(zeev): call batches manager only if it's a batch in the first place.
```

Comment:
```suggestion
```
@ItayLevyOfficial @yaelAbergel

Reply:
[handled in an open pr](https://github.com/dwallet-labs/dwallet-network/pull/408/commits/74c9758ed8ad319bb07705a33287f65a17b56f2f)